File: ItalianRestaurantFrontend/src/app/models/user-dto.ts
export class UserDto {
  firstName: string;
  lastName: string;
  email: string;
  phoneNumber: string;
  newsletter: boolean;

  constructor(firstName: string, lastName: string, email: string, phoneNumber: string, newsletter: boolean) {
    this.firstName = firstName;
    this.lastName = lastName;
    this.email = email;
    this.phoneNumber = phoneNumber;
    this.newsletter = newsletter;
  }
}

File: ItalianRestaurantBackend/src/main/java/com/example/italianrestaurant/user/UserDto.java
package com.example.italianrestaurant.user;

import jakarta.validation.constraints.Email;
import jakarta.validation.constraints.NotBlank;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
public class UserDto {
    @NotBlank(message = "Firstname is mandatory")
    private String firstName;
    @NotBlank(message = "Lastname is mandatory")
    private String lastName;
    @NotBlank(message = "Email is mandatory")
    @Email
    private String email;
    @NotBlank(message = "Phone number is mandatory")
    private String phoneNumber;
    private boolean newsletter;
}
